netbout-web/src/main/java/com/netbout/rest/bout/TkIndex.java

```diff
@@ -83,75 +84,81 @@ public Response act(final Request req) throws IOException {
+                            new XeTransform.Func<Attachment>() {
+                                @Override
+                                public XeSource transform(final Attachment atmt)
+                                    throws IOException {
+                                    return new XeAttachment(
+                                        request, bout, atmt
+                                    );
+                                }
                             }
-                        }
-                    )
-                ),
-                new XeAppend(
-                    "messages",
-                    new XeTransform<>(
-                        messages(bout, request, query),
-                        new XeTransform.Func<Message>() {
-                            @Override
-                            public XeSource transform(final Message msg)
-                                throws IOException {
-                                return new XeMessage(bout, msg);
+                        )
+                    ),
+                    new XeAppend(
+                        "messages",
+                        new XeTransform<>(
+                            messages(bout, request, query),
+                            new XeTransform.Func<Message>() {
+                                @Override
+                                public XeSource transform(final Message msg)
+                                    throws IOException {
+                                    return new XeMessage(bout, msg);
+                                }
                             }
-                        }
+                        )
                     )
-                )
-            ),
-            new XeAppend("query", query),
-            new XeLink("post", home.path("post")),
-            new XeLink("preview", home.path("preview")),
-            new XeLink("rename", home.path("rename")),
-            new XeLink("invite", home.path("invite")),
-            new XeLink("search", home.path("search")),
-            new XeLink("upload", home.path("upload")),
-            new XeLink("create", home.path("create")),
-            new XeLink("attach", home.path("attach")),
-            new XeLink("subscribe", home.path("subscribe"))
-        );
+                ),
+                new XeAppend("query", query),
+                new XeLink("post", home.path("post")),
+                new XeLink("preview", home.path("preview")),
+                new XeLink("rename", home.path("rename")),
+                new XeLink("invite", home.path("invite")),
+                new XeLink("search", home.path("search")),
+                new XeLink("upload", home.path("upload")),
+                new XeLink("create", home.path("create")),
+                new XeLink("attach", home.path("attach")),
+                new XeLink("subscribe", home.path("subscribe"))
+            );
+        } catch (final Inbox.BoutNotFoundException ex) {
```

Comment:
@2686747 did you test that this is necessary somehow? To me this looks like a really big change.
Couldn't we simply throw this instead then in the above code?:

```java
throw new HttpException(HttpURLConnection.HTTP_NOT_FOUND, new BoutNotFoundException(this.bout))
```
We shouldn't be wrapping this whole set of directives for sure. This exception would definitely be caught in the normal way and we're already doing the same in `com.netbout.rest.bout.RqBout#bout`.
I think this would be a lot safer especially since we're not adding a test.